app/R/app_utilities.R

```diff
@@ -0,0 +1,367 @@
+    y_axis_title <- "Effective Marginal Tax Rate"
+  }
+  
+  
+  data1_for_plot[, value1 := pmax(pmin(value1, max_rate), min_rate)]
+  data1_for_plot <- 
+    data1_for_plot[, .(gross_wage1, gross_wage1_annual, value1)]
+  
+  data1_for_plot %<>% melt(id.vars=c('gross_wage1','gross_wage1_annual'),
+                           variable.name='Scenario')
+  
+  data1_for_plot[Scenario=="value1", Scenario:=policy_name1]
+  
+  data1_for_plot %>% dcast(gross_wage1 + gross_wage1_annual ~ Scenario) %>%
+    plot_ly(x = ~gross_wage1_annual, y = ~`Status Quo`, name = policy_name1, 
+            mode = "lines", type = 'scatter',
+            line = list(color = "#56B4E9", width = 3),
+            hovertemplate =  paste0(
+              "Annual gross wage income:\n %{x:$,.2f}\n ",
+              policy_name1, ": %{y:.2%} <extra></extra>")) %>% 
+    add_trace(x=~hours1, y=~0, line=list(width = 0), xaxis="x2", 
+              data=data1, showlegend=FALSE, inherit=FALSE,
+              hoverinfo="none", type = "scatter", mode = "lines") %>%
+    layout(xaxis2 = list(overlaying = "x", nticks = 10, side = "top",
+                         title = "Hours/week", automargin=TRUE, size=8,
+                         showline = TRUE),
+           xaxis = list(title = "Annual gross wage income ($)", 
+                        tickformat =  "$,",
+                        automargin=TRUE,
+                        showline = TRUE,
+                        mirror=TRUE,
+                        margin = list(b = 0)),
+           yaxis = list (title = y_axis_title,
+                         # texttemplate='%{y:.2%}',
+                         tickformat = ".0%",
+                         automargin=TRUE,
+                         showline = TRUE,
+                         mirror=TRUE,
+                         range = c(0, max(1, data1_for_plot$value) + 0.03)
+                         ),
+           legend = list(x = 100, y = 0.5),
+           hovermode = "compare") 
+}
+
+remove_IWTC_from_params <- function(input_params) {
+  output_params <- copy(input_params)
+  output_params$FamilyAssistance_IWTC_Rates_UpTo3Children <- 0
+  output_params$FamilyAssistance_IWTC_Rates_SubsequentChildren <- 0
+  return(output_params)
+}
+
+choose_IWTC_or_benefit <- function(X, X_without_IWTC) {
+  # Merge max
+  SQ_net_income_comparison <- cbind(
+    X[, .(With_IWTC = Net_Income)],
+    X_without_IWTC[, .(Without_IWTC = Net_Income)]
+  )
+  SQ_net_income_comparison[, row_ID := 1:.N]
+  With_IWTC_indices <- SQ_net_income_comparison[With_IWTC >= Without_IWTC, row_ID]
+  Without_IWTC_indices <- SQ_net_income_comparison[Without_IWTC > With_IWTC, row_ID]
+  
+  With_IWTC <- X[With_IWTC_indices]
+  Without_IWTC <- X_without_IWTC[Without_IWTC_indices]
+  
+  X <- rbind(With_IWTC, Without_IWTC)
+  setorderv(X, "hours1")
+  
+  X[, EMTR := 1 - 1L*(shift(Net_Income,1L,type="lead")-Net_Income)]
+  X[, EMTR := zoo::na.locf(EMTR)]
+  
+  return(X)
+}
+
+# plotly - income composition
```

Comment:
I don't think you need this function any more? It seems like amounts_net_plot covers it now.